abstract_specification.md

```diff
@@ -591,6 +591,12 @@ This is a pre-release specification in active development. As defined by [semver
 
 _Note:_ this API was preceded by another (un-versioned) API draft, which is colloquially referred to as `0.0.0-dev` and `0.1.0-dev`
 
+## Slicing
+
+* Slices are doubly inclusive -- `slice(2,4)` includes array indices 2, 3, and 4.
+* Slices may be doubly open -- `slice(None)` means select all
+* Slices may be half-open -- `slice(2,None)` on a five-element array selects indices 2, 3, and 4; `slice(None, 2)` selects indices 0, 1, and 2.
```

Comment:
The syntax here is somewhat Python-specific. It may be worth specifying this more generally and saying that the general concept can be represented as a Python `slice`.